File: week_2/problem_set_2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
  // Prompt the user for some text input
  string text = get_string("Text: ");

  // Count words, letters, and sentences
  int words = 0;
  int letters = 0;
  int sentences = 0;

  for (int i = 0, n = strlen(text); i <= n; i++)
  {
    if (65 <= toupper(text[i]) && toupper(text[i]) <= 90)
    {
      letters++;
    }
    else if (text[i] == 46 || text[i] == 33 || text[i] == 63)
    {
      sentences++;
    }
    else if (text[i] == 32 || text[i] == '\0')
    {
      words++;
    }
  }

  // Calculate L and S
  float L = letters * 100.00 / words;
  float S = sentences * 100.00 / words;

  // Compute Coleman-Liau index
  int index = round(0.0588 * L - 0.296 * S - 15.8);

  // Return message
  if (index < 1)
  {
    printf("Before Grade 1\n");
  }
  else if (index > 16)
  {
    printf("Grade 16+\n");
  }
  else
  {
    printf("Grade %i\n", index);
  }
}

File: week_06/problem_set_6/05_readability/readability.py
# Prompt the user for input
text = input("Text: ")

# Define variables
words = len(text.rsplit())
letters = 0
sentences = 0

# Count
for character in text:
    if 65 <= ord(character.upper()) <= 90:
        letters += 1
    elif ord(character) in [33, 46, 63]:
        sentences += 1

# Calculate L and S
L = letters * 100.00 / words
S = sentences * 100.00 / words

# Calculate index
index = round(0.0588 * L - 0.296 * S - 15.8)

# Evaluate condition and print
if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")
